contracts/poap/src/execute.rs

```diff
@@ -0,0 +1,409 @@
+    /// * `users` - List of users for whom the POAP will be minted.
+    pub fn mint_to(
+        &self,
+        deps: DepsMut,
+        _env: Env,
+        info: MessageInfo,
+        users: Vec<String>,
+        extension: T,
+    ) -> Result<Response<C>, ContractError> {
+        self.assert_is_minter(deps.storage, &info.sender)?;
+
+        let mut minted_tokens = Vec::<String>::with_capacity(users.len());
+        for user in users {
+            let user_addr = deps.api.addr_validate(&user)?;
+            minted_tokens.push(self.mint_to_user(deps.storage, &user_addr, extension.clone())?);
+        }
+
+        Ok(Response::new()
+            .add_attribute("action", "mint_to")
+            .add_attribute("minter", info.sender)
+            .add_attribute("token_ids", minted_tokens.join(", ")))
+    }
+}
+
+// Utility functions
+impl<'a, T, C, E, Q> PoapContract<'a, T, C, E, Q>
+where
+    T: Serialize + DeserializeOwned + Clone + Debug,
+    C: CustomMsg,
+    E: CustomMsg,
+    Q: CustomMsg,
+{
+    /// Computes the id of the next POAP to mint.
+    pub fn generate_poap_id(&self, storage: &dyn Storage) -> StdResult<String> {
+        Ok(format!("{}", 1 + self.cw721_base.token_count(storage)?))
+    }
+
+    /// Mint a POAP to an user.
+    /// * `owner` - User for whom the POAP will be minted.
+    pub fn mint_to_user(
+        &self,
+        storage: &mut dyn Storage,
+        owner: &Addr,
+        extension: T,
+    ) -> Result<String, ContractError> {
+        // Create the token
+        let token = cw721_base::state::TokenInfo {
+            owner: owner.clone(),
+            approvals: vec![],
+            token_uri: Some(self.metadata_uri.load(storage)?),
+            extension,
+        };
+
+        // Generate the token id
+        let token_id = self.generate_poap_id(storage)?;
+        self.cw721_base
+            .tokens
+            .update(storage, &token_id, |old| match old {
+                Some(_) => Err(ContractError::Claimed {}),
+                None => Ok(token),
+            })?;
+
+        self.cw721_base.increment_tokens(storage)?;
+
+        Ok(token_id)
+    }
+
+    /// Asserts that the provided address is the contract admin.
+    /// * `sender` - Address that will be checked.
+    pub fn assert_is_admin(
+        &self,
+        storage: &dyn Storage,
+        sender: &Addr,
+    ) -> Result<(), ContractError> {
+        cw_ownable::assert_owner(storage, sender).map_err(|e| ContractError::Ownership(e))
+    }
+
+    /// Asserts that the provided address is the contract minter.
+    /// * `sender` - Address that will be checked.
+    pub fn assert_is_minter(
+        &self,
+        storage: &dyn Storage,
+        sender: &Addr,
+    ) -> Result<(), ContractError> {
+        let minter = self.minter.load(storage)?;
+        if !sender.eq(&minter) {
+            return Err(ContractError::MintUnauthorized {});
+        }
```

Comment:
We need to allow the contract admin to act as a minter. If the `minter` is not set, nobody, including the admin, will be able to mint tokens. On the contrary, with this new check, if the `minter` is set, only that specified person can mint tokens. To ensure this functionality works correctly, we should add tests for both cases:
1. When the `minter` is specified, the admin can mint some tokens.
2. When the `minter` is not specified, the admin can still mint some tokens.